graph/network/maximum_flow.go

```diff
@@ -0,0 +1,168 @@
+package network
+
+import (
+	"container/list"
+	"fmt"
+	"math"
+
+	"gonum.org/v1/gonum/graph"
+	"gonum.org/v1/gonum/graph/simple"
+)
+
+type edgeKey struct{ from, to int64 }
```

Comment:
Structure code so that caller is placed before callee (and the equivalent in type use). This gives context for the reader to understand where/how a defined function or type will be used, and we are not writing C, so we are not constrained by its source ordering.